src/frontend/context/ManglendeBrevVariablerContext.ts

```diff
@@ -0,0 +1,44 @@
+import { useState } from 'react';
+
+import constate from 'constate';
+
+import { MalStruktur, Valg, Valgfelt, Variabel } from '../komponenter/Brev/typer';
+import { harIkkeVerdi } from '../utils/utils';
+
+export const [ManglendeBrevVariablerProvider, useManglendeBrevVariabler] = constate(() => {
+    const [manglendeBrevVariabler, settManglendeBrevVariabler] = useState<Variabel[]>([]);
+
+    const finnManglendeBrevVariabler = (
+        mal: MalStruktur,
+        inkluderteDelmaler: Record<string, boolean>,
+        valgfelt: Partial<Record<string, Record<Valgfelt['_id'], Valg>>>,
+        variabler: Partial<Record<string, string>>
+    ): Variabel[] => {
+        return mal.delmaler
+            .filter((delmal) => inkluderteDelmaler[delmal._id])
+            .flatMap((delmal) => Object.values(valgfelt[delmal._id] ?? {}))
+            .filter((valg) => valg._type === 'tekst')
+            .flatMap((valg) => valg.variabler)
+            .filter((variabel) => harIkkeVerdi(variabler[variabel._id]));
+    };
+
+    const oppdaterManglendeBrevVariabler = (
+        mal: MalStruktur,
+        inkluderteDelmaler: Record<string, boolean>,
+        valgfelt: Partial<Record<string, Record<Valgfelt['_id'], Valg>>>,
+        variabler: Partial<Record<string, string>>
+    ) => {
+        return settManglendeBrevVariabler(
+            finnManglendeBrevVariabler(mal, inkluderteDelmaler, valgfelt, variabler)
+        );
+    };
+
+    const brevMalManglerVariabler = (): boolean => {
+        return manglendeBrevVariabler.length !== 0;
+    };
+    return {
+        manglendeBrevVariabler,
+        oppdaterManglendeBrevVariabler,
+        brevMalManglerVariabler,
+    };
```

Comment:
Trenger ingen funksjon for å sjekke om length er 0

```suggestion
    const brevManglerVariabler = manglendeBrevVariabler.length !== 0;
    return {
        manglendeBrevVariabler,
        oppdaterManglendeBrevVariabler,
        brevManglerVariabler,
    };
```

Reply:
92659e8b3e5f2c89e7bc2b561f44b74ee790c388